drivers/ksfilter/ks/irp.c

```diff
@@ -369,15 +370,27 @@ KsQueryInformationFile(
             /* use FastIoQueryBasicInfo routine */
             if (FastIoDispatch->FastIoQueryBasicInfo)
             {
-                return FastIoDispatch->FastIoQueryBasicInfo(FileObject, TRUE, (PFILE_BASIC_INFORMATION)FileInformation, &IoStatusBlock, DeviceObject);
+                Result = FastIoDispatch->FastIoQueryBasicInfo(FileObject, TRUE, (PFILE_BASIC_INFORMATION)FileInformation, &IoStatusBlock, DeviceObject);
+
+                if (Result && NT_SUCCESS(IoStatusBlock.Status))
+                {
+                    /* request was handled and succeeded */
+                    return STATUS_SUCCESS;
```

Comment:
Then maybe it could be better to return the result of IoStatusBlock.Status ...
(this remark is based on the behaviour of
https://git.reactos.org/?p=reactos.git;a=blob;f=ntoskrnl/io/iomgr/iofunc.c;hb=e2b536bcb120422085766b56af79ffbc883383a7#l835
and
https://git.reactos.org/?p=reactos.git;a=blob;f=ntoskrnl/io/iomgr/iofunc.c;hb=e2b536bcb120422085766b56af79ffbc883383a7#l2396
)

Reply:
Done, that is what I was referring to ;-)